compiler/parser/RdrHsSyn.hs

```diff
@@ -2614,58 +2673,69 @@ mkTemplateInstanceMethods templateName =
       let bodyName = mkUnqualVar $ mkVarOcc $ prefixTemplateClassMethod $ methodName ++ templateName
       in  mkTemplateClassMethod methodName [] bodyName Nothing
 
+withInstanceContext :: String -> [Located RdrName] -> LHsType GhcPs -> HsType GhcPs
+withInstanceContext templateName tyVars =
+  let tInstanceClass = mkUnqualClass $ templateName ++ "Instance"
```

Comment:
We should have one single place where we do this `++ "Instance"` thing. We might want to change it to `++ ":Instance"` or something similar in the future and it would be great if we only need to change one place.

Reply:
Good point